snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/StateManager.kt

```diff
@@ -0,0 +1,187 @@
+package com.snowplowanalytics.core.tracker
+
+import com.snowplowanalytics.snowplow.event.AbstractSelfDescribing
+import com.snowplowanalytics.snowplow.event.Event
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+import com.snowplowanalytics.snowplow.tracker.InspectableEvent
+import java.util.*
+
+class StateManager {
+    
+    private val identifierToStateMachine = HashMap<String, StateMachineInterface>()
+    private val stateMachineToIdentifier = HashMap<StateMachineInterface, String>()
+    private val eventSchemaToStateMachine = HashMap<String?, MutableList<StateMachineInterface>>()
```

Comment:
Just adding a note as I came across this file – there is no need to have the schema key an nullable string, I think we can remove the question mark here as well in other places in the file that deal with schemas and the schema registry. Generally, the less nullables we have, the better (so we don't have to handle them and to make the code easier to reason about).